edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Mqtt/MqttUsernameParser.cs

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace Microsoft.Azure.Devices.Edge.Hub.Mqtt
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using Microsoft.Azure.Devices.Edge.Hub.Core;
+    using Microsoft.Azure.Devices.Edge.Util;
+
+    public class MqttUsernameParser : IUsernameParser
+    {
+        const string ApiVersionKey = "api-version";
+        const string DeviceClientTypeKey = "DeviceClientType";
+
+        public (string deviceId, string moduleId, string deviceClientType) Parse(string username)
```

Comment:
nit: it might be out of scope of this change, but it would be nice to change return type to Option<> and get rid of exceptions in this method :)

Reply:
yes, but it was an already existing code I moved out for reuse.